sync/src/relayer/mod.rs

```diff
@@ -215,10 +215,11 @@ impl<CS: ChainStore + 'static> Relayer<CS> {
         }
     }
 
-    pub fn accept_block(&self, nc: &CKBProtocolContext, peer: PeerIndex, block: &Arc<Block>) {
-        let ret = self.chain.process_block(Arc::clone(&block), true);
-
-        if ret.is_ok() {
+    pub fn accept_block(&self, nc: &CKBProtocolContext, peer: PeerIndex, block: Block) {
+        if self
+            .shared()
+            .insert_new_block(&self.chain, peer, block.clone())
```

Comment:
This should be an Arc<block>.

Reply:
Updated. Change the parameters of `insert_new_block` to `block: Arc<Block>`